spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/config/ConformanceConfigInstance.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.conformance.config
+
+import org.apache.spark.storage.StorageLevel
+import scopt.OParser
+import za.co.absa.enceladus.common.config.{ConfigError, JobConfig}
+import za.co.absa.enceladus.dao.auth.{InvalidMenasCredentialsFactory, MenasCredentialsFactory}
+
+import scala.util.Try
+
+
+/**
+ * This is a class for configuration provided by the command line parameters
+ *
+ * Note: scopt requires all fields to have default values.
+ *       Even if a field is mandatory it needs a default value.
+ */
+case class ConformanceConfigInstance(datasetName: String = "",
+                                     datasetVersion: Int = 1,
+                                     reportDate: String = "",
+                                     reportVersion: Option[Int] = None,
+                                     menasCredentialsFactory: MenasCredentialsFactory = InvalidMenasCredentialsFactory,
+                                     performanceMetricsFile: Option[String] = None,
+                                     folderPrefix: Option[String] = None,
+                                     persistStorageLevel: Option[StorageLevel] = None,
+                                     publishPathOverride: Option[String] = None,
+                                     experimentalMappingRule: Option[Boolean] = None,
+                                     isCatalystWorkaroundEnabled: Option[Boolean] = None,
+                                     autocleanStandardizedFolder: Option[Boolean] = None,
+                                     credsFile: Option[String] = None,
+                                     keytabFile: Option[String] = None
+                                    )
+  extends ConformanceConfig[ConformanceConfigInstance] with JobConfig[ConformanceConfigInstance] {
```

Comment:
For readability, I would
1. remove all the `{}` line 51 vs 59
2. Move copy statement always to the next line

Reply:
2. Seems a bit too much for me

Reply:
It provides a consistency I would say